### PR TITLE
0.5.0-RC1 Blog Post

### PR DESCRIPTION
**Pull request by allanrenucci**
Here is a draft of 0.5.0-RC1 blog post.
- @nicolasstucki Do you want to say something about `dotc -decompile` or `dotc -from-tasty`?
- @Duhemm Do you want to say something about java signature? Maybe give examples of what it lets you do with reflection.

### CODE REVIEW

File: docs/blog/_posts/2017-11-21-fifth-dotty-milestone-release.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+## What’s new in the 0.5.0-RC1 technology preview?
+
+### Reworked implicit search [#3421](https://github.com/lampepfl/dotty/pull/3421)
+TODO @odersky
+
+### TASTY frontend
+The TASTY frontend is a frontend that reloads the ASTs from the TASTY within the classfiles in the classpath. We fixed most issues with the frontend which enable us to re-compile code from TASTY without the need of retypecheck (#3436, #3512, #3492). 
+
+Currenlty two backends that use this frontend are available
```
Comment (OlivierBlanvillain) [inline docs/blog/_posts/2017-11-21-fifth-dotty-milestone-release.md]: Currently

File: docs/blog/_posts/2017-11-21-fifth-dotty-milestone-release.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+* Dotty class file decompiler `dotc -decompile -classpath CLASSPATH CLASSNAME`. It will decompile the code by preaty printing the contents of the TASTY. This code is the high level representation of the typechecked code without loss of precieceness due transformations towards JVM bytecode. It can also be used with `dotc -decompile -print-tasty ...` to show a preaty printed version of the raw TASTY (#3519).
```
Comment (OlivierBlanvillain) [inline docs/blog/_posts/2017-11-21-fifth-dotty-milestone-release.md]: s/preaty/pretty/g

Comment (OlivierBlanvillain) [inline docs/blog/_posts/2017-11-21-fifth-dotty-milestone-release.md]: preciseness

**Comment by Duhemm** [comment]
@allanrenucci I added a short paragraph about generic java signatures.

**Comment by allanrenucci** [comment]
@heathermiller Could you please have a look? I'll open a PR to scala-lang if we are happy with this version

File: docs/blog/_posts/2017-11-29-fifth-dotty-milestone-release.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+and the query `implicitly[C]`.
+
+This query would now be classified as ambiguous. This makes sense, after all
+there are two possible solutions, `b(a1)` and `b(a2)`, neither of which is better
+than the other and both of which are better than the third solution, `c`.
+By contrast, Scala 2 would have rejected the search for `A` as
+ambiguous, and subsequently have classified the query `b(implictly[A])` as a normal fail,
+which means that the alternative `c` would be chosen as solution!
+
+Scala 2's somewhat puzzling behavior wrt ambiguity has been exploited to implement the analogue
```
Comment (OlivierBlanvillain) [inline docs/blog/_posts/2017-11-29-fifth-dotty-milestone-release.md]: s/wrt/with respect to/

**Comment by OlivierBlanvillain** [comment]
:+1:

File: docs/blog/_posts/2017-11-29-fifth-dotty-milestone-release.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+of a "negated" search in implicit resolution, where a query `Q1` fails if some other
+query `Q2` succeeds and `Q1` succeeds if `Q2` fails. With the new cleaned up behavior these
+techniques no longer work. But there is now a new special type `scala.implicits.Not` which
+implements negation directly. For any query type `Q`: `Not[Q]` succeeds if and only if the
+implicit search for `Q` fails.
+
+### Dependent function types [#3464](https://github.com/lampepfl/dotty/pull/3464)
+A dependent function type describes functions where the result type may depend
```
[the remark this answers is not in the document]
Comment (allanrenucci) [reply docs/blog/_posts/2017-11-29-fifth-dotty-milestone-release.md]: @odersky What do you think?

**Pull request merged**